src/storage/datasets_index.rs

```diff
@@ -43,206 +44,62 @@ impl DatasetsIndex {
                 .collect()
         })
     }
-
-    pub fn get_headers(&self) -> &reqwest::header::HeaderMap {
-        &self.http_headers
-    }
-}
-
-// Reuses memory building both ChunkSet and DatasetsIndex simultaneously
-pub fn parse_assignment(assignment: WorkerAssignment) -> anyhow::Result<(ChunkSet, DatasetsIndex)> {
-    let mut datasets = HashMap::new();
-    let mut chunk_set = ChunkSet::new();
-
-    let filename_refs: HashMap<u32, Arc<str>> = assignment
-        .known_filenames
-        .into_iter()
-        .enumerate()
-        .map(|(i, filename)| (i as u32, Arc::from(filename.as_str())))
-        .collect();
-
-    for DatasetChunks {
-        dataset_id,
-        download_url,
-        chunks,
-        ..
-    } in assignment.dataset_chunks
-    {
-        let dataset_id: Arc<Dataset> = Arc::from(dataset_id);
-        let mut dataset_files = HashMap::new();
-        for chunk in chunks {
-            let data_chunk = DataChunk::from_str(&chunk.path)
-                .map_err(|e| anyhow::anyhow!("Invalid chunk path: {e}"))?;
-            let files = chunk
-                .filenames
-                .iter()
-                .map(|index| filename_refs.get(index).unwrap().clone())
-                .collect();
-            chunk_set.insert(ChunkRef {
-                dataset: dataset_id.clone(),
-                chunk: data_chunk.clone(),
-            });
-            dataset_files.insert(data_chunk, files);
+    pub fn from(assigned_data: Vec<crate::util::assignment::Dataset>, headers: BTreeMap<String, String>) -> Self {
+        let mut datasets = HashMap::new();
+        let mut chunks_ordinals_map = HashMap::new();
+        let mut ordinal = 0;
+        for dataset in assigned_data {
+            let dataset_id = dataset.id;
+            let dataset_url = dataset.base_url;
+            let mut dataset_files: HashMap<DataChunk, Vec<String>> = Default::default();
+            for chunk in dataset.chunks {
+                let data_chunk = DataChunk::from_path(&chunk.id).unwrap();
+                let mut files: Vec<String> = Default::default();
+                for (file, _) in chunk.files {
```

Comment:
Why is url ignored? Let's leave a TODO to implement it later

Reply:
TODO left